src/Annotations/Serializer/CommonMark/functions.php

```diff
@@ -2,9 +2,25 @@
 
 namespace eLife\Annotations\Serializer\CommonMark;
 
-function clean_paragraph($text)
+use League\CommonMark\Util\Xml;
+
+function clean_paragraph(string $text) : string
 {
     $allowed_tags = '<i><sub><sup><span><del><math><a><br><caption>';
 
     return strip_tags($text, $allowed_tags);
 }
+
+function escape_math(string $text) : string
+{
+    // Escape MathML.
+    $escaped = preg_replace_callback('~(?P<before><math[^>]*>)(?P<mathml>.*)(?P<after></math>)~s', function ($match) {
```

Comment:
I wonder if the regex is greedy and what happens with multiple <math> tags. I'll ensure there is a test for that.

Reply:
I will add a test.

Reply:
Named groups aren't needed? Could be simplified to:

```php
function escape_math(string $text) : string
{
    $escape = function (array $matches) {
        return Xml::escape($matches[0]);
    };

    return preg_replace_callback_array([
        '~(<math[^>]*>.*?</math>)~s' => $escape,
        '~(\$\$.+?\$\$)~s' => $escape,
    ], $text);
}
```